esmvaltool/cmorizers/obs/cmorize_obs_lai3g.py

```diff
@@ -0,0 +1,216 @@
+"""ESMValTool CMORizer for LAI3g data.
+
+Tier
+    Tier 3: restricted dataset.
+
+Source
+    http://cliveg.bu.edu/modismisr/lai3g-fpar3g.html
+
+Last access
+    20190503
+
+Download and processing instructions
+    To obtain the data sets it is necessary to contanct Ranga B. Myneni
+    (Department of Earth and Environment, Boston University). See link above
+    for more information.
+
+    By default, this dataset is regridded to a 1°x1° grid (original resoultion
+    is 1/12°). If you want to use the original resolution, remove the `regrid`
+    section in the configuration file (`LAI3g.yml`). Note that in this case,
+    preprocessing the dataset with ESMValTool (i.e. every time you run the
+    tool) can take a very long time (> 30 min).
+
+"""
+
+import fnmatch
+import logging
+import os
+import shutil
+import zipfile
+from datetime import datetime
+
+import iris
+import iris.coord_categorisation
+import numpy as np
+from cf_units import Unit
+
+from esmvalcore.preprocessor import regrid
+
+from . import utilities as utils
+
+logger = logging.getLogger(__name__)
+
+# Properties of the binary file (cannot be stored in .yml since computations
+# are necessary)
+DTYPE = '>i2'
+N_LAT = 2160
+N_LON = 4320
+MISSING_VALUE = -32768
+SCALE_FACTOR = 1000.0
+UPPER_LEFT_LAT = 90.0 - 1.0 / 24.0
+UPPER_LEFT_LON = -180.0 + 1.0 / 24.0
+LOWER_RIGHT_LAT = -90.0 + 1.0 / 24.0
+LOWER_RIGHT_LON = 180.0 - 1.0 / 24.0
+MONTHS = {
+    'jan': 1,
+    'feb': 2,
+    'mar': 3,
+    'apr': 4,
+    'may': 5,
+    'jun': 6,
+    'jul': 7,
+    'aug': 8,
+    'sep': 9,
+    'oct': 10,
+    'nov': 11,
+    'dec': 12,
+}
+DAYS = {
+    'a': 8,
+    'b': 23,
+}
+
+
+def _clean(file_dir):
+    """Remove unzipped input files."""
+    if os.path.isdir(file_dir):
+        shutil.rmtree(file_dir)
+        logger.info("Removed cached directory %s", file_dir)
+
+
+def _extract_variable(cmor_info, attrs, in_dir, out_dir, cfg):
```

Comment:
Maybe you could split this into a few more functions? E.g. a function that creates the `nc_files`